### DISCUSSION TITLE
Problem with bunding a 'StyledProperty' to a variable

### DISCUSSION THREAD

**Discussion by doronweiss** [AvaloniaUI/Avalonia#16126] [General]
i shoetenned my usercontrol class:
`  public partial class LayerSelector : UserControl {

    public static readonly StyledProperty<bool> checkedProperty =
      AvaloniaProperty.Register<LayerSelector, bool>(nameof(Checked), defaultValue: true);

    public bool Checked {
      get => (bool)GetValue(checkedProperty);
      set => SetValue(checkedProperty, (bool)value);
    }

    public LayerSelector() {
      InitializeComponent();
      this.DataContext = this;
    }
  }`
It has a style property, i insert my control in a window, if i bind the property to a const value it works fine:
`        <local:LayerSelector Checked="True"/> 
`
But if i try to bind it to a *variable*:  
`
 <local:LayerSelector Text="Tiles" Checked=**"{Binding DispTiles}"** AColor="Red" />
`
i get a compiletime error:  
> Avalonia error AVLN:0004: Unable to find suitable setter or adder for property Checked of type DetPageTest:DetPageTest.LayerSelector

this binding works for non user defined properties

**Comment by stevemonaco** [comment]
You should try changing your `checkedProperty` to `CheckedProperty`. It may be required convention.

**Reply by doronweiss** [reply]
You are RIGHT!  i would have never guessed that b y myself still it solves the issue, thanks
